evennia/web/api/filters.py

```diff
@@ -0,0 +1,27 @@
+from django_filters.rest_framework.filterset import FilterSet
+
+from evennia.objects.models import ObjectDB
+from evennia.accounts.models import AccountDB
+from evennia.scripts.models import ScriptDB
+
+SHARED_FIELDS = ["db_key", "db_typeclass_path", "db_tags__db_key", "db_tags__db_category"]
```

Comment:
If I remember right, you can define custom filters that map to arbitrary querystrings, so I should be able to do this

Reply:
This is a bit more complex than I thought since django-filters don't support complex queries by default: what we want is `?alias=foo` doing `Q(db_tags__db_tagtype="alias") & Q(db_tags__db_key="foo")`. I think this is probably doable, but it's not something their documentation covers as far as I can tell. I'll go through django-filter's code a bit and see what I can pick out.

Reply:
Maybe it would be useful to just use key but to embed the alias-search within the key search as well? That is after all how other searching works (you give a search term and both keys and aliases are searched together).

Reply:
django-filter works by letting you specify specific GET arguments to be passed along as filtering: it's not like a search bar where there's just one argument. The idea is that for a front-end form you might have a search box that could have drop-downs, radio buttons, or whatever in addition to a CharField, and those arguments all need to be handled separately to further constrain results. As an example from django admin, think of it as not just the search bar, but also the filters you can set on the right side of the admin window.

I did come up with a way to do it by letting them specify an ?aliases or ?permissions argument and establishing a pattern that should be easy enough to extend. It'd be like this:
`mygame.com/api/objects?aliases="blue"&permissions="builder"` to search for any objects that have an alias of 'blue' and builder permissions.